application/helpers/frontend_helper.php

```diff
@@ -1886,13 +1889,39 @@ function doAssessment($surveyid, $returndataonly=false)
                             if ($usresult)
                             {
                                 $usrow = $usresult->read();
-                                $fieldmap[$field['fieldname']]['assessment_value']=$usrow['assessment_value'];
-                                $total=$total+$usrow['assessment_value'];
+                                $assessmentValue = $usrow['assessment_value'];
                             }
                         }
                     }
                     $groups[]=$field['gid'];
                 }
+
+                // If this is a question (and not a survey field, like ID), save asessment value
+                if (!empty($field['title']))
+                {
+                    /**
+                     * Allow Plugin to update assessment value
+                     */
+                    // Prepare Event Info
+                    $event = new PluginEvent('afterSurveyQuestionAssessment');
+                    $event->set('surveyId', $surveyid);
+                    $event->set('question', $field);
```

Comment:
qid is the id of the question : we send qid for 'this question' : the parent question. Send same information for plugin dev please.
Here : we have to send qid (parent qid) + other informations for sub(s)-question(s).

here field come from fielmap : we are REALLY unsure if we kee fieldmap in 4/5 ... release